service/resource/endpoint/current.go

```diff
@@ -0,0 +1,50 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+
+	pod, err := toPod(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	podAnnotations := pod.GetAnnotations()
+	podIP, ipOk := podAnnotations["endpoint.kvm.giantswarm.io/ip"]
+	podServiceName, serviceNameOk := pod.Annotations["endpoint.kvm.giantswarm.io/service"]
+
+	if !(ipOk && serviceNameOk) {
```

Comment:
Either one of these will be false if the Annotation does not exist.

Reply:
This is a nice example of how to ease control structures and separate concerns of business logic implementations. I think it would be nice to have a helper method that takes a custom object or pod and returns the information held by the annotations. In case the information cannot be found the helper method returns an error. We can check for this error and cancel the context in case it is not nil. Then we can just return an error within the helper method as soon as some annotation is not found and we can prevent such complicated conditionals.

Reply:
I changed this into a function :+1: